c7n/resources/ami.py

```diff
@@ -129,10 +134,90 @@ def process(self, images):
+                filters:
+                  - type: image-age
+                    days: 30
+                actions:
+                  - set-deprecation
+                    #Number of days from AMI creation
+                    age: 90
+                    #Number of days from now
+                    #days: 90
+                    #Specific date/time
+                    #date: "{now}"
+
+    """
+
+    schema = type_schema(
+        'set-deprecation',
+        date={'type': 'string'},
+        days={'type': 'integer'},
+        age={'type': 'integer'})
+    permissions = ('ec2:EnableImageDeprecation', 'ec2:DisableImageDeprecation')
+    dep_date = None
+    dep_age = None
+
+    def validate(self):
+        try:
+            if 'date' in self.data:
+                self.dep_date = parse_date(self.data.get('date'))
+                if not self.dep_date:
+                    raise PolicyValidationError(
+                        "policy:%s filter:%s has invalid date format" % (
+                            self.manager.ctx.policy.name, self.type))
+            elif 'days' in self.data:
+                self.dep_date = (datetime.datetime.now(tz=tzutc()) +
+                    timedelta(days=int(self.data.get('days'))))
+            elif 'age' in self.data:
+                self.dep_age = (int(self.data.get('age')))
+        except ValueError:
+            raise PolicyValidationError(
+                "policy:%s filter:%s has invalid time interval" % (
+                    self.manager.ctx.policy.name, self.type))
+
+    def process(self, images):
+        client = local_session(self.manager.session_factory).client('ec2')
+        image_count = len(images)
+        images = [i for i in images if self.manager.ctx.options.account_id == i['OwnerId']]
+        if len(images) != image_count:
+            self.log.info("Implicitly filtered %d non owned images", image_count - len(images))
+        for i in images:
+            if not self.dep_date and not self.dep_age:
+                self.manager.retry(client.disable_image_deprecation, ImageId=i['ImageId'])
+            else:
+                if self.dep_age:
+                    date = parse_date(i['CreationDate']) + timedelta(days=self.dep_age)
+                else:
+                    date = self.dep_date
+                # Hack because AWS won't let you set a deprecation time in the
+                # past - set to now + 1 minute if the time is in the past
+                if date < datetime.datetime.now(tz=tzutc()):
+                    date = datetime.datetime.now(tz=tzutc()) + timedelta(minutes=1)
+                self.manager.retry(client.enable_image_deprecation,
+                    ImageId=i['ImageId'], DeprecateAt=date)
+
+
 @AMI.action_registry.register('remove-launch-permissions')
 class RemoveLaunchPermissions(BaseAction):
     """Action to remove the ability to launch an instance from an AMI
 
+    DEPRECATED - use set-permissions instead to support AWS Organizations
```

Comment:
we have a deprecation framework that would be good to use here, its not used to much yet, but this is a great fit for it, `c7n/deprecated.py` it will report deprecated syntax in cli validate, and optionally error based on flags.